File: tutionmediaapp/lib/models/teacher_profile.dart
class TeacherProfile {
  final int id;
  final int userId;
  final String? fullName;
  final String? phone;
  final String? qualification;
  final String? institution;
  final int? experienceYears;
  final String? preferredClasses;
  final String? preferredSubjects;
  final String? location;
  final String? bio;
  final bool isVerified;

  TeacherProfile({
    required this.id,
    required this.userId,
    this.fullName,
    this.phone,
    this.qualification,
    this.institution,
    this.experienceYears,
    this.preferredClasses,
    this.preferredSubjects,
    this.location,
    this.bio,
    required this.isVerified,
  });

  factory TeacherProfile.fromJson(Map<String, dynamic> json) {
    return TeacherProfile(
      id: json['id'],
      userId: json['user_id'],
      fullName: json['full_name'],
      phone: json['phone'],
      qualification: json['qualification'],
      institution: json['institution'],
      experienceYears: json['experience_years'],
      preferredClasses: json['preferred_classes'],
      preferredSubjects: json['preferred_subjects'],
      location: json['location'],
      // expected_salary removed
      bio: json['bio'],
      isVerified: (json['is_verified'] == 1 || json['is_verified'] == true) ? true : false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'full_name': fullName,
      'phone': phone,
      'qualification': qualification,
      'institution': institution,
      'experience_years': experienceYears,
      'preferred_classes': preferredClasses,
      'preferred_subjects': preferredSubjects,
      'location': location,
      'bio': bio,
      'is_verified': isVerified,
    };
  }
}

File: backend/models/TeacherProfile.js
const pool = require('../config/database');

class TeacherProfile {
  static async createOrUpdate(userId, profileData) {
    const { full_name, phone, qualification, institution, experience_years, preferred_classes, preferred_subjects, location, bio, is_verified = false } = profileData;
    const [result] = await pool.execute(
      `INSERT INTO teacher_profiles (user_id, full_name, phone, qualification, institution, experience_years, preferred_classes, preferred_subjects, location, bio, is_verified)
       VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
       ON DUPLICATE KEY UPDATE
       full_name = VALUES(full_name),
       phone = VALUES(phone),
       qualification = VALUES(qualification),
       institution = VALUES(institution),
       experience_years = VALUES(experience_years),
       preferred_classes = VALUES(preferred_classes),
       preferred_subjects = VALUES(preferred_subjects),
       location = VALUES(location),
       bio = VALUES(bio),
       is_verified = VALUES(is_verified)`,
      [userId, full_name, phone, qualification, institution, experience_years, preferred_classes, preferred_subjects, location, bio, is_verified]
    );
    return result;
  }

  static async findByUserId(userId) {
    const [rows] = await pool.execute('SELECT * FROM teacher_profiles WHERE user_id = ?', [userId]);
    return rows[0];
  }
}

module.exports = TeacherProfile;